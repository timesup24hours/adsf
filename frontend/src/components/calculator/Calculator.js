import React from 'react'

const Calculator = props => {
  return (
    <div className='Calculator-container'>
      <div className='Calculator-button'>
        <button
          className='waves-effect waves-light btn blue'
          onClick={() => console.log('Calculator.js place order implementation needed!')} >
          Place your order
        </button>
      </div>
      <div className='Calculator-holder'>
        <div className='Calculator-div Calculator-order-summary'>Order Summary</div>
        <div className='Calculator-div'>
          <div>Items ({props.cart.totalQuantity}):</div>
          <div>$ {props.cart.total}</div>
        </div>
        <div className='Calculator-div Calculator-shipping-handling'>
          <div>Shipping & handling:</div>
          <div>$ </div> {/* shipping fee calcu function needed */}
        </div>
        <div className='Calculator-divider'>
          <div></div>
        </div>
        <div className='Calculator-div'>
          <div>Total before tax:</div>
          <div>$ </div> {/* total + shipping */}
        </div>
        <div className='Calculator-div'>
          <div>Estimated tax to be collected:</div>
          <div>$ </div> {/* tax */}
        </div>
        <div className='Calculator-divider'>
          <div></div>
        </div>
        <div className='Calculator-div Calculator-total-summary'>
          <div>Order Total:</div>
          <div>$</div> {/* order total */}
        </div>
      </div>
    </div>
  )
}

export default Calculator
