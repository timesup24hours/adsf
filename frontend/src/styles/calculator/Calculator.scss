.Calculator-container {
  max-width: 300px;
  min-width: 300px;
  height: 320px;
  display: flex;
  flex-direction: column;
  padding: 14px 16px;
  font-size: 14px;
  box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, .1);
}
.Calculator-holder {
  margin-top: 6px;
  padding: 0px 4px;
}
.Calculator-button {
  button {
    width: 100%;
  }
}
.Calculator-order-summary {
  font-weight: bold;
  font-size: 16px;
}

.Calculator-div {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 6px 0px;
}
.Calculator-divider {
  margin: 10px 0px;
  div {
    border-bottom: 1px solid grey;
  }
}
.Calculator-total-summary {
  font-weight: bold;
  color: red;
  font-size: 16px;
}
