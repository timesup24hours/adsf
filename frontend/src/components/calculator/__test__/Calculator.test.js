import Calculator from '../Calculator';

test('# Calculator', () => {
  const wrapper = shallow(<Calculator cart={{ totalQuantity: '1', total: '1' }} />)
  expect(wrapper).toMatchSnapshot();
})
