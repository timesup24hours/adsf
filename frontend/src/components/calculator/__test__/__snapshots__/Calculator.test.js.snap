exports[`test # Calculator 1`] = `
<div
  className="Calculator-container">
  <div
    className="Calculator-button">
    <button
      className="waves-effect waves-light btn blue"
      onClick={[Function]}>
      Place your order
    </button>
  </div>
  <div
    className="Calculator-holder">
    <div
      className="Calculator-div Calculator-order-summary">
      Order Summary
    </div>
    <div
      className="Calculator-div">
      <div>
        Items (
        1
        ):
      </div>
      <div>
        $ 
        1
      </div>
    </div>
    <div
      className="Calculator-div Calculator-shipping-handling">
      <div>
        Shipping & handling:
      </div>
      <div>
        $ 
      </div>
       
    </div>
    <div
      className="Calculator-divider">
      <div />
    </div>
    <div
      className="Calculator-div">
      <div>
        Total before tax:
      </div>
      <div>
        $ 
      </div>
       
    </div>
    <div
      className="Calculator-div">
      <div>
        Estimated tax to be collected:
      </div>
      <div>
        $ 
      </div>
       
    </div>
    <div
      className="Calculator-divider">
      <div />
    </div>
    <div
      className="Calculator-div Calculator-total-summary">
      <div>
        Order Total:
      </div>
      <div>
        $
      </div>
       
    </div>
  </div>
</div>
`;
